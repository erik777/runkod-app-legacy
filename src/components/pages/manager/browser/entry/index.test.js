import React from "react";
import TestRenderer from "react-test-renderer";


import {Icon} from './index';

describe('Browser Entry - Icon', () => {

  test('1- jpeg icon', () => {
    const props = {
      type: 'image/jpeg'
    };

    const renderer = TestRenderer.create(
      <Icon {...props}/>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  test('2- png icon', () => {
    const props = {
      type: 'image/png'
    };

    const renderer = TestRenderer.create(
      <Icon {...props}/>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  test('3- html icon', () => {
    const props = {
      type: 'text/html'
    };

    const renderer = TestRenderer.create(
      <Icon {...props}/>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  test('4- json icon', () => {
    const props = {
      type: 'application/json'
    };

    const renderer = TestRenderer.create(
      <Icon {...props}/>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });


  test('5- javascript icon', () => {
    const props = {
      type: 'text/javascript'
    };

    const renderer = TestRenderer.create(
      <Icon {...props}/>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  test('6- css icon', () => {
    const props = {
      type: 'text/css'
    };

    const renderer = TestRenderer.create(
      <Icon {...props}/>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });


  test('7- unknown type icon', () => {
    const props = {
      type: 'text/php'
    };

    const renderer = TestRenderer.create(
      <Icon {...props}/>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});