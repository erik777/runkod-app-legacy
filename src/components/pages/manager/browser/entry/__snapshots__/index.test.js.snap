// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browser Entry - Icon 1- jpeg icon 1`] = `
<div
  className="entry-icon entry-icon-file icon-image"
>
  <div
    className="shape"
  >
    <div
      className="type-name"
    >
      img
    </div>
  </div>
</div>
`;

exports[`Browser Entry - Icon 2- png icon 1`] = `
<div
  className="entry-icon entry-icon-file icon-image"
>
  <div
    className="shape"
  >
    <div
      className="type-name"
    >
      img
    </div>
  </div>
</div>
`;

exports[`Browser Entry - Icon 3- html icon 1`] = `
<div
  className="entry-icon entry-icon-file icon-html"
>
  <div
    className="shape"
  >
    <div
      className="type-name"
    >
      &lt;/&gt;
    </div>
  </div>
</div>
`;

exports[`Browser Entry - Icon 4- json icon 1`] = `
<div
  className="entry-icon entry-icon-file icon-json"
>
  <div
    className="shape"
  >
    <div
      className="type-name"
    >
      { }
    </div>
  </div>
</div>
`;

exports[`Browser Entry - Icon 5- javascript icon 1`] = `
<div
  className="entry-icon entry-icon-file icon-js"
>
  <div
    className="shape"
  >
    <div
      className="type-name"
    >
      js
    </div>
  </div>
</div>
`;

exports[`Browser Entry - Icon 6- css icon 1`] = `
<div
  className="entry-icon entry-icon-file icon-css"
>
  <div
    className="shape"
  >
    <div
      className="type-name"
    >
      css
    </div>
  </div>
</div>
`;

exports[`Browser Entry - Icon 7- unknown type icon 1`] = `
<div
  className="entry-icon entry-icon-file"
>
  <div
    className="shape"
  >
    
  </div>
</div>
`;
